actors/builtin/multisig/multisig_test.go

```diff
@@ -75,10 +78,208 @@ func TestConstruction(t *testing.T) {
+		var nilParams = abi.MethodParams{}
+
+		rt := builder.WithBalance(abi.NewTokenAmount(0), abi.NewTokenAmount(0)).Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustProposeWithExitCode(rt, chuck, value, method, nilParams, exitcode.ErrInsufficientFunds)
+
+		actor.mustContainTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    abi.NewTokenAmount(value),
+			Method:   abi.MethodNum(method),
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("fail propose from non-signer", func(t *testing.T) {
+		// non-signer address
+		richard := newSecpAddr(t, 105)
+		const unlockDuration = int64(0)
+		const numApprovals = int64(2)
+		const method = int64(0)
+		const value = int64(10)
+
+		var signers = []addr.Address{anne, bob}
+		var nilParams = abi.MethodParams{}
+
+		rt := builder.Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.SetCaller(richard, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustProposeWithExitCode(rt, chuck, value, method, nilParams, exitcode.ErrForbidden)
+
+		actor.mustContainTransactions(rt)
+	})
+}
+
+//
+// Helper methods for calling multisig actor methods
+//
+
+type msActorHarness struct {
+	multisig.MultiSigActor
+	t testing.TB
+}
+
+func (h *msActorHarness) mustConstructMultisigActor(rt *mock.Runtime, numApprovalsThresh, unlockDuration int64, signers ...addr.Address) {
+	constructParams := multisig.ConstructorParams{
+		Signers:               signers,
+		NumApprovalsThreshold: numApprovalsThresh,
+		UnlockDuration:        abi.ChainEpoch(unlockDuration),
+	}
+
+	rt.ExpectValidateCallerAddr(builtin.InitActorAddr)
+	constructRet := h.MultiSigActor.Constructor(rt, &constructParams)
+	require.Equal(h.t, runtime.EmptyReturn{}, *constructRet)
+	rt.Verify()
+}
+
+func (h *msActorHarness) mustPropose(rt *mock.Runtime, to addr.Address, value, method int64, params abi.MethodParams) {
+	proposeParams := &multisig.ProposeParams{
+		To:     to,
+		Value:  abi.NewTokenAmount(value),
+		Method: abi.MethodNum(method),
+		Params: params,
+	}
+	h.MultiSigActor.Propose(rt, proposeParams)
+	rt.Verify()
+}
+
+func (h *msActorHarness) mustProposeWithExitCode(rt *mock.Runtime, to addr.Address, value, method int64, params abi.MethodParams, exitCode exitcode.ExitCode) {
+	proposeParams := &multisig.ProposeParams{
+		To:     to,
+		Value:  abi.NewTokenAmount(value),
+		Method: abi.MethodNum(method),
+		Params: params,
+	}
+
+	rt.ExpectAbort(exitCode, func() {
+		h.MultiSigActor.Propose(rt, proposeParams)
+	})
+
+	rt.Verify()
+}
+
+func (h *msActorHarness) mustContainTransactions(rt *mock.Runtime, expected ...multisig.MultiSigTransaction) {
```

Comment:
`assertTransactions` is a bit more clear they must exactly match, not just contain as a subset.